src/tls_stream.rs

```diff
@@ -200,6 +201,35 @@ pub enum HandshakeError<S> {
     Interrupted(MidHandshakeTlsStream<S>),
 }
 
+/// A struct used to wrap various cert chain validation results for callback processing. 
+pub struct CertValidationResult {
+    chain :CertChainContext,
+    res :io::Result<()>,
+    chain_index :i32,
+    element_index :i32,
+    extra_policy_status : (winapi::LPCSTR, *const winapi::c_void),
```

Comment:
To clarify: 
I only meant to describe which struct fields could potentially be relevant for us.
Those particular fields might be relevant for some use case in the future 
(when we would access them and expose them without breaking BC).

Since we currently aren't even aware of a use case,
there's no need to keep them around.